CRM/Core/BAO/FinancialTrxn.php

```diff
@@ -39,31 +39,27 @@ public static function create($params) {
     $trxn = new CRM_Financial_DAO_FinancialTrxn();
     $trxn->copyValues($params);
 
-    if (isset($params['fee_amount']) && is_numeric($params['fee_amount'])) {
-      if (!isset($params['total_amount'])) {
-        $trxn->fetch();
-        $params['total_amount'] = $trxn->total_amount;
-      }
-      $trxn->net_amount = $params['total_amount'] - $params['fee_amount'];
+    if (!isset($trxn->total_amount)) {
+      $trxn->fetch();
     }
+    $trxn->net_amount = ($trxn->total_amount ?? 0) - ($trxn->fee_amount ?? 0);
 
-    if (empty($params['id']) && !CRM_Utils_Rule::currencyCode($trxn->currency)) {
+    if (empty($trxn->id) && !CRM_Utils_Rule::currencyCode($trxn->currency)) {
       $trxn->currency = CRM_Core_Config::singleton()->defaultCurrency;
```

Comment:
If $params['id'] was passed in, it got copied to $trxn via `$trxn->copyValues()`. Historically `$params` was passed by reference but it is not anymore.